plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt

```diff
@@ -194,40 +201,119 @@ class GradleEnclavePlugin @Inject constructor(private val layout: ProjectLayout)
         }
     }
 
+    private fun createGenerateGramineSGXManifestTask(
+        target: Project,
+        enclaveFatJarTask: Jar,
+        generateGramineManifestTask: GenerateGramineDirectManifest,
+        enclaveExtension: EnclaveExtension,
+        createDummyKeyTask: GenerateDummyMrsignerKey,
+        type: BuildType
+    ): GenerateGramineSGXManifest {
+        return target.createTask("generateSGXGramineManifest$type") { task ->
+            val directManifestPath = generateGramineManifestTask.manifestFile.get().asFile.absolutePath
+            val outputSgxManifestPath = "$directManifestPath.sgx"
+            val outputTokenPath = directManifestPath.replace("manifest", "token")
+            val outputSig = directManifestPath.replace("manifest", "sig")
+
+            task.privateKey.set(enclaveExtension.signingType.flatMap {
+                when (it) {
+                    SigningType.DummyKey -> createDummyKeyTask.outputKey
+                    SigningType.PrivateKey -> enclaveExtension.signingKey
+                    else -> target.provider { null }
+                }
+            })
+            task.enclaveJar.set(enclaveFatJarTask.archiveFile)
+            if (pythonSourcePath != null) {
+                val pythonFiles = target.fileTree(pythonSourcePath).files
+                if (pythonFiles.size == 1) {
+                    task.pythonSourcePath.set(pythonFiles.first()!!)
+                } else {
+                    throw GradleException(
+                        "Only a single Python script is supported, but ${pythonFiles.size} were " +
+                                "found in $pythonSourcePath"
+                    )
+                }
+            }
+            task.directManifest.set(generateGramineManifestTask.manifestFile)
+            task.sgxManifest.set(Paths.get(outputSgxManifestPath).toFile())
+            task.sgxToken.set(Paths.get(outputTokenPath).toFile())
+            task.sgxSig.set(Paths.get(outputSig).toFile())
```

Comment:
I think a single `OutputDir` would be better, but it's fine for now. I think I might end up changing this as part of the improvements to the plugin tests I'm doing on the side